chipy_org/apps/job_board/views.py

```diff
@@ -11,12 +11,14 @@
 from chipy_org.apps.job_board.forms import JobPostForm, JobUserForm, JobProfileForm
 from .models import JobPost
 
+
 @login_required
 def create_job_post(request):
 
     if request.method == "POST":
 
```

Comment:
Just wondering if you looked at this documentation [long_required](https://docs.djangoproject.com/en/2.2/topics/auth/default/#the-login-required-decorator).  

I see you were able to get this to work with the updates to chipy_org/templates/_login_options.html.  Does this work with python-social-auth?   Maybe a more obvious way as it keep the redirect closer to the view that requires the login.  No need to update.

Reply:
Yes, I did look at that document. I'm not sure what the alternative you are suggesting. **Do you think you can tell me in code?**

Our settings.py defines a `LOGIN_REDIRECT_URL = "/" `and `LOGIN_URL = "/login/"`

 When you try to visit the `chipy.org/job/create/` , it redirects you to login at  `www.chipy.org/login/?next=/job/create/` . This is okay so far. But then it deals with Social Auth.  The `_login_options.html ` somehow tells the browser to go to the home page, instead of where we want to go, which is `/job/create/ `.  After logging in, Social Auth redirects to `chipy.org/#` if you're logging in with Google.

So Social Auth is defining the ultimate destination of where the user ends up.  Please read the first section "Return the user to the original page" in this link
https://python-social-auth.readthedocs.io/en/latest/use_cases.html
This document uses `?next={{ request.path }}` but that would take one to the `'www.chipy.org' `site since that is **path**.  The answer that worked for me is `?next={{ request.GET.next }}`.

Reply:
Sorry, I misread the document I referenced above and erroneously pointed it out. 

I thought we could just pass the redirect url as an argument to the login_required decorator but that only customize the "next" field name in the context passed to the template.  

Thanks for pointing out the social auth doc.  I wonder why {request.path} didn't work as documented, but I have no issues as I tested it and it does work as intended.    

Kudos on the job board!